ui/perfherder/alerts/StatusDropdown.jsx

```diff
@@ -257,38 +258,98 @@ export default class StatusDropdown extends React.Component {
+    const differenceInTime = Math.abs(secondDate - firstDate);
+    let differenceInDays = Math.ceil(differenceInTime / (1000 * 60 * 60 * 24));
+
+    // If weekday of date1 is bigger than date2 then it is a new week
+    // and we have to subtract the weekend days
+    if (firstDate.getUTCDay() > secondDate.getUTCDay()) {
+      differenceInDays -= 2;
+    } else {
+      differenceInDays -= 2 * Math.trunc(differenceInDays / 7);
+    }
+
+    let differenceInHours = Math.ceil(differenceInTime / (1000 * 60 * 60));
+    // If due date is Monday and today is Friday and we have to show the hours left,
+    // we have to subtract the weekend hours from the difference
+    if (firstDate.getUTCDay() === 5 && secondDate.getUTCDay() === 1) {
+      differenceInHours -= 2 * 24;
+    }
+
+    return {
+      hours: differenceInHours,
+      days: differenceInDays,
+    };
+  }
+
+  calculateCountdowns(now, dueDate, difference) {
+    if (difference.hours < 24 && difference.hours >= 0) {
+      return `${difference.hours} hours left`;
+    }
+
+    if (now.getTime() >= dueDate.getTime()) {
+      return `Overdue`;
+    }
+
+    return `${difference.days} days left`;
+  }
+
   displayDueDateCountdown() {
+    const { alertSummary } = this.props;
+
     const now = new Date(Date.now());
     const currentDay = now.getDay();
     const saturday = 6;
     const sunday = 0;
-    const dueDate = new Date(this.props.alertSummary.triage_due_date);
+    const triageDueDate = new Date(alertSummary.triage_due_date);
+    const bugDueDate = new Date(alertSummary.bug_due_date);
 
-    const differenceInTime = Math.abs(dueDate - now);
-    const differenceInDays = Math.ceil(
-      differenceInTime / (1000 * 60 * 60 * 24),
-    );
-    const differenceInHours = Math.ceil(differenceInTime / (1000 * 60 * 60));
+    const triageDifference = this.getRemainingTime(now, triageDueDate);
+    const bugDifference = this.getRemainingTime(now, bugDueDate);
+
+    const countdowns = {
+      triage: '',
+      bug: '',
+    };
 
     // if the website is accessed during the weekend, nothing will be shown
     if (currentDay === saturday || currentDay === sunday) {
       this.showCountdownToTriageIcon = false;
-      return '';
+      return countdowns;
     }
-
-    if (now.getDate() === dueDate.getDate()) {
-      return `Hours left: ${differenceInHours}`;
+    if (alertSummary.first_triaged) {
+      countdowns.bug = this.calculateCountdowns(now, bugDueDate, bugDifference);
+    } else {
+      countdowns.triage = this.calculateCountdowns(
+        now,
+        triageDueDate,
+        triageDifference,
+      );
+      countdowns.bug = this.calculateCountdowns(now, bugDueDate, bugDifference);
     }
 
-    if (now.getTime() >= dueDate.getTime()) {
-      return `Overdue`;
-    }
+    return countdowns;
+  }
 
-    if (differenceInDays >= 4) {
-      return `Days left: ${differenceInDays - 2}`;
+  calculateTriageCountdown(dueDateStatus) {
```

Comment:
For consistency I would keep `countdowns` or `dueDateStatus` in the code, but not both, the logic is already hard to follow.

Reply:
Was this addressed? Could you point me to the file where I can find the change? Because of all the recent changes, it's hard to follow where it was addressed.